src/modules/sv_detection_methods/analyze_sa_tag_method.cpp

```diff
@@ -57,29 +56,82 @@ void retrieve_aligned_segments(std::string const & sa_string, std::vector<Aligne
 void analyze_aligned_segments(std::vector<AlignedSegment> const & aligned_segments,
                               std::vector<Junction> & junctions,
                               seqan3::dna5_vector const & query_sequence,
-                              std::string const & read_name)
+                              std::string const & read_name,
+                              int32_t const min_length,
+                              int32_t const max_overlap)
 {
-    for(size_t i = 1; i<aligned_segments.size(); i++)
+    for (size_t i = 1; i<aligned_segments.size(); i++)
     {
         AlignedSegment current = aligned_segments[i-1];
         AlignedSegment next = aligned_segments[i];
         int32_t distance_on_read = next.get_query_start() - current.get_query_end();
-        // Check that there is neither an overlap nor a larger gap on the read
-        // TODO(eldarion): add command-line parameters for changing these currently hard-coded cutoffs
-        if (distance_on_read >= 0 && distance_on_read <= 10)
+        // Check that the overlap between two consecutive alignment segments
+        // of the read is lower than the given threshold
+        if (distance_on_read >= -max_overlap)
         {
-            Breakend mate1{current.ref_name,
-                           current.orientation == strand::forward ? current.get_reference_end()
-                                                                  : current.get_reference_start(),
-                           current.orientation};
-            Breakend mate2{next.ref_name,
-                           next.orientation == strand::forward ? next.get_reference_start()
-                                                               : next.get_reference_end(),
-                           next.orientation};
-            auto inserted_bases = query_sequence | seqan3::views::slice(current.get_query_end(), next.get_query_start());
-            Junction new_junction{mate1, mate2, inserted_bases, read_name};
-            seqan3::debug_stream << "BND: " << new_junction << "\n";
-            junctions.push_back(std::move(new_junction));
+            int32_t mate1_pos;
+            if (current.orientation == strand::forward)
+            {
+                mate1_pos = current.get_reference_end();
+            }
+            else
+            {
+                mate1_pos = current.get_reference_start();
+            }
```

Comment:
You can shorten this, if you want:
```suggestion
            if (current.orientation == strand::forward)
                mate1_pos = current.get_reference_end();
            else
                mate1_pos = current.get_reference_start();
```